05/Textarea.js

```diff
@@ -1,8 +1,39 @@
 import React from 'react';
 
 class Textarea extends React.Component {
+
     render() {
-        return <textarea></textarea>
+        return <textarea style={{height:'31px'}} onChange={this.handleTextarea} ref={el => this.textarea = el}></textarea>
+    }
+
+    handleTextarea = () => {
+        const {addText} = this.props;
+        const newText = this.textarea.value;
+        addText(newText);
+    }
+
+    getSnapshotBeforeUpdate(prevProps,prevState){
+        const height = this.textarea.offsetHeight;
+        const scroll = this.textarea.scrollHeight
+        console.log('height:' + height);
+        console.log('scroll:' + scroll);
+
+        if(height < scroll || height <= 100){
```

Comment:
Tutaj jest zły warunek, powinien być `&&` (oby dwa muszą być spełnione) - musisz na to uważać, bo to już drugi raz w ostatnim czasie ;P